[18.0][ADD] account_payment_batch_oca_reconcile

alexis-via opened the pull request:
<img width="1303" height="673" alt="reconcile_payment_lot" src="https://github.com/user-attachments/assets/65af06a3-5074-4527-ad1f-258241f044f7" />

etobella committed:
[ADD] account_reconcile_payment_order

alexis-via committed:
Rename account_reconcile_payment_order to account_payment_batch_oca_reconcile

alexis-via committed:
[MIG] account_payment_batch_oca_reconcile

Migrate module account_payment_batch_oca_reconcile using the new object
account.payment.lot now available in account_payment_batch_oca.

alexis-via committed:
[IMP] account_payment_batch_oca_reconcile: remove unneeded code

etobella reviewed (changes requested):
History is not well preserved

alexis-via commented:
I'm happy to announce that the OCA/bank-payment-alternative core contributors have decided that the 19.0 branch (and future branches) will be created from the 18.0 branch, thus keeping the full history even for renamed modules (same method as used by OCA/mis-builder).
This history has been preserved on this PR. And, with the decision taken for the 19.0+ branches, I will continue to be preserved on the next odoo versions.
@etobella I'm fully commited to keep the history of the commits, you can trust me on this.

etobella commented:
Nope, it is decided by PSC and IMO it should require a complete agreement and I don't agree.

So no

pedrobaeza commented:
I'm curious who are "OCA/bank-payment-alternative core contributors", as that's not any official group.

alexis-via committed:
[IMP] account_payment_batch_oca_reconcile: fr translation

dreispt commented:
Hi, PSC decision making was mentioned, so I'd like to share this piece of information about that:
https://github.com/OCA/odoo-community.org/blob/master/the-association/Organization.rst#53decision-making

Thank you.

etobella commented:
Thanks @dreispt Good to hear that. Thanks for sharing as I didn't know we had this. In any case, if we want to make this change, we need some PSC voting for this.

So with this in mind alternative shouldn't exist? (3 negatives and 2 positives)...

dreispt commented:
This is actually what the board uses for decisions, but I also learned recently we had it written down.
For me the most important part comes after that:

> The rules require that a negative vote includes an alternative proposal or a detailed explanation of the reasons for the negative vote.
>
>The community then tries to gather consensus on an alternative proposal that resolves the issue. In the great majority of cases, the concerns leading to the negative vote can be addressed.

etobella commented:
For what?

The creation of an alternative? We are creating a fork just for a decision on a field. Can't we stick together somehow? Or is it with me or against me? We offered to help to converge several times, but we just asked 1 thing, keeping the old mode for some time until odoo ensures the real usage of it. We offered options...

Change of behavior in the repo? It is only used in two repos with strong maintainers behind. We agreed in other meetings that this change should be approved by the PSC team. Will alexis maintain the modukes in the same way for all versions (we all know that he skips some of them)? Or it is just a decision because he doesn't want to listen to our comments or follow the (according to his own words) "stupid rules of OCA"? Also, the change is complex for external users, as they see the module there and expect it to work, but it is not working yet. Also, when a change happens in the previous versions and the module has not been migrated it is easy to lose the change. That is a problem that has happened in the past. For me the main reason for not doing it. 

Is this enough reasons for both decisions? Do I need to offer more reasons?

pedrobaeza commented:
@dreispt and what about self-merging with no reviews the PRs (all of them, even big changes or new modules) of this repository:

https://github.com/OCA/intrastat-extrastat/pull/258#issuecomment-1935708946
https://github.com/OCA/bank-payment-alternative/pull/24#issuecomment-3120591874

And all the guidelines (for example of commit messages - https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#commit-message) not being followed?

I'm very tired of this situation.

alexis-via commented:
> History is not well preserved

I rewrote the module name in the first commit, using the command you suggested in one of your comments on https://github.com/OCA/bank-payment-alternative/pull/5

hparfr commented on account_payment_batch_oca_reconcile/models/account_bank_statement_line.py:
```diff
@@ -0,0 +1,95 @@
+        In this case we need to add all the liquidity lines. Otherwise, we remove them
+        """
+        lot = self.add_payment_lot_id
+        if lot:
```
```suggestion
        if not lot:
            return
```

hparfr commented on account_payment_batch_oca_reconcile/models/account_bank_statement_line.py:
```diff
@@ -0,0 +1,95 @@
+                    self.manual_reference,
+                )
+                ["payment_lot_counterparts"].append(lot.id)
+                counterparts = set(data_info["payment_lot_counterparts"])
```
```suggestion
                counterparts = set(data_info["payment_lot_counterparts"])
```
please use another variable name or aligne the type with counterparts = [] above

hparfr commented on account_payment_batch_oca_reconcile/models/account_bank_statement_line.py:
```diff
@@ -0,0 +1,95 @@
+        result = super()._recompute_suspense_line(
+            data, reconcile_auxiliary_id, manual_reference
+        )
+        result["payment_lot_counterparts"] = payment_lot_counterparts
```
```suggestion
        result = super()._recompute_suspense_line(
            data, reconcile_auxiliary_id, manual_reference
        )
        payment_lot_counterparts = (
            self.reconcile_data_info
            and self.reconcile_data_info.get("payment_lot_counterparts", [])
        ) or []
        result["payment_lot_counterparts"] = payment_lot_counterparts
```

hparfr commented on account_payment_batch_oca_reconcile/models/account_payment_lot.py:
```diff
@@ -0,0 +1,65 @@
+    @api.depends("payment_ids.state")
+    def _compute_matched_info(self):
+        for lot in self:
+            lot.is_matched = any([pay.state == "paid" for pay in lot.payment_ids])
```
```suggestion
            lot.is_matched = "paid" in lot.payment_ids.mapped('state')
```

alexis-via committed:
[IMP] account_payment_batch_oca_reconcile: support 'Add all' button

Add support for "Add all" button in 'Payment lot' tab
Code clean-up suggested by Raphaël Reverdy on it's code review

hparfr commented on account_payment_batch_oca_reconcile/models/account_payment_lot.py:
```diff
@@ -0,0 +1,65 @@
+        valid_account_types = self.env[
+            "account.payment"
+        ]._get_valid_payment_account_types()
+        move_lines = self.env["account.move.line"]
```
```suggestion
        move_lines = self.env["account.move.line"].browse(False)
```
To better show the intent

hparfr reviewed (approved):
comments taken in account
Thanks @alexis-via

alexis-via commented:
@etobella I rewrote yesterday the commit history as you suggested. Could you update your "change request" ?

alexis-via commented:
@etobella As you can see here https://github.com/OCA/bank-payment-alternative/pull/32/commits/8ccd9020bb8fcaf63fa8f9c24de813102f45cb86  I rewrote the commit history using the technique you suggested. Could you please update your "change request" ? Thanks !

alexis-via commented:
@etobella As I said in my previous comments, I implemented your change request and I executed the git command you suggested on this PR:
```
git filter-branch -f --index-filter '
git ls-files --stage | \
sed "s:account_reconcile_payment_order/:account_payment_batch_oca_reconcile
/:" | \
GIT_INDEX_FILE=$GIT_INDEX_FILE.new \
git update-index --index-info && \
mv "$GIT_INDEX_FILE.new" "$GIT_INDEX_FILE"
' HEAD
```
Could you update your change request please?

sbidoul commented on account_payment_batch_oca_reconcile/models/account_bank_statement_line.py:
```diff
@@ -0,0 +1,105 @@
+        "account.payment.lot",
+        check_company=True,
+        prefetch=False,
+    )
```
A little comment or help text explain the purpose of this field would help.

alexis-via replied:
I added a comment to explain the role of this field

alexis-via commented:
As the change request of @etobella  has been implemented and he didn't update his change request, I will merge this, as we have 2 positive reviews.

alexis-via commented:
/ocabot merge nobump
